ui/plugin-system/src/model/panels.tsx

```diff
@@ -36,13 +36,20 @@ export interface PanelProps<Options extends JsonObject> {
 }
 
 /**
- * Renders a PanelComponent from a panel plugin at runtime.
+ * Hook for using a panel plugin at runtime.
  */
-export const PanelComponent: PanelPlugin['PanelComponent'] = (props) => {
-  const plugin = usePlugin('Panel', props.definition.kind);
-  if (plugin === undefined) {
-    return null;
+export function usePanelPlugin(kind: string): PanelPlugin<JsonObject> {
+  const plugin = usePlugin('Panel', kind);
+  if (plugin !== undefined) {
+    return plugin;
   }
-  const { PanelComponent: PluginComponent } = plugin;
-  return <PluginComponent {...props} />;
+
+  // Return a default/placeholder plugin while loading happens
+  return defaultPanelPlugin;
```

Comment:
This is just code golf, but I wouldn't mind putting the defaultPanelPlugin that returns when the panel is undefined and loading higher in the component, and returning the real panel at the end of the ocmponent. 

But whatever, feel free ignore this / only use if you agree too.

Reply:
Yeah, I don't love this either. I think I want to make some changes to plugin loading to make it explicit that plugins could be undefined while they're loading. There's too much "magic" going on behind the scenes right now to avoid violating the rules of hooks that lead to some kinda unexpected behaviors. Returning a "default" plugin definitely doesn't feel quite right.